packages/compiler/core/cli.ts

```diff
@@ -745,14 +745,19 @@ function run(command: string, commandArgs: string[], options?: RunOptions) {
 }
 
 function internalCompilerError(error: unknown): never {
-  // NOTE: An expected error, like one thrown for bad input, shouldn't reach
-  // here, but be handled somewhere else. If we reach here, it should be
-  // considered a bug and therefore we should not suppress the stack trace as
-  // that risks losing it in the case of a bug that does not repro easily.
-  console.error("Internal compiler error!");
-  console.error("File issue at https://github.com/microsoft/cadl");
-  console.error();
-  console.error(error);
+  if (error instanceof ExternalError) {
+    console.error(error);
```

Comment:
Maybe worth a comment here about how ExternalError already has the appropriate content.